### PR TITLE
FIX-#2285: Default to pandas warning message improved

### LINKED ISSUE #2285: Improve default to pandas warning messages
Currently, these warnings don't allow us to understand which method causes the problem. 

```
UserWarning: <function DefaultMethod.call.<locals>.applyier at 0x7d5ca99d28b0> defaulting to pandas implementation.
To request implementation, send an email to feature_requests@modin.org.
UserWarning: <function DefaultMethod.call.<locals>.applyier at 0x7d5ca996e3a0> defaulting to pandas implementation.
```

Can we improve it?

### PR DESCRIPTION
**Pull request by dchigarev**
Signed-off-by: Dmitry Chigarev <dmitry.chigarev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2285 
- [x] tests added and passing

Example of improved warning:
```
# How it started:
UserWarning: <function DefaultMethod.call.<locals>.applyier at 0x7f8a004300d0> defaulting to pandas implementation.

# How it's going:
UserWarning: <function GroupBy.prod> defaulting to pandas implementation.

```

### CODE REVIEW

File: modin/pandas/test/test_general.py [hunk 1]
```diff
@@ -437,3 +438,29 @@ def test_to_pandas_indices():
...
+@pytest.mark.skipif(
+    get_current_backend() != "BaseOnPython",
+    reason="This test make sense only on BaseOnPython backend.",
+)
+@pytest.mark.parametrize(
+    "func, regex",
+    [
+        (lambda df: df.mean(level=0), r"[^D]*DataFrame\.mean"),
+        (lambda df: df + df, r"[^D]*DataFrame\.add"),
+        (lambda df: df.index, r"[^D]*DataFrame.get_axis\(0\)"),
```
Comment (anmyachev) [inline modin/pandas/test/test_general.py:452]: Missed?

```suggestion
        (lambda df: df.index, r"[^D]*DataFrame\.get_axis\(0\)"),
```

File: modin/pandas/test/test_general.py [hunk 2]
```diff
@@ -437,3 +438,29 @@ def test_to_pandas_indices():
...
+
+@pytest.mark.skipif(
+    get_current_backend() != "BaseOnPython",
+    reason="This test make sense only on BaseOnPython backend.",
+)
+@pytest.mark.parametrize(
+    "func, regex",
+    [
+        (lambda df: df.mean(level=0), r"[^D]*DataFrame\.mean"),
+        (lambda df: df + df, r"[^D]*DataFrame\.add"),
```
Comment (anmyachev) [inline modin/pandas/test/test_general.py:451]: Why you using regexp like `[^D]` ?

File: modin/pandas/test/test_general.py (hunk 1 above)
Comment (dchigarev) [reply modin/pandas/test/test_general.py]: yeah, thanks

File: modin/pandas/test/test_general.py (hunk 2 above)
Comment (dchigarev) [reply modin/pandas/test/test_general.py]: yeah, you right, `[^D]*` is redundant, removed that

**Review by anmyachev — APPROVED** [review]
LGTM!

**Pull request merged**